sdk/formrecognizer/azure-ai-formrecognizer/migration-guide.md

```diff
@@ -0,0 +1,255 @@
+        System.out.printf("Document field: %s%n", name);
+        System.out.printf("Document field type: %s%n", documentFieldSchema.getType().toString());
+        System.out.printf("Document field confidence: %.2f%n", docTypeInfo.getFieldConfidence().get(name));
+    });
+});
+```
+
+### Manage models
+In 3.X, listing models returned only the custom trained models using the `listCustomModel` method.
+With 4.X, list Models operation `listModels` now returns a paged list of prebuilt in addition to custom models and does not return in-progress and failed models.
+<!-- embedme ./src/samples/java/com/azure/ai/formrecognizer/ReadmeSamples.java#L322-L351 -->
+```java
+AtomicReference<String> modelId = new AtomicReference<>();
+
+// First, we see how many models we have, and what our limit is
+AccountProperties accountProperties = documentModelAdminClient.getAccountProperties();
+System.out.printf("The account has %s models, and we can have at most %s models",
+    accountProperties.getDocumentModelCount(), accountProperties.getDocumentModelLimit());
+
+// Next, we get a paged list of all of our models
+PagedIterable<DocumentModelInfo> customDocumentModels = documentModelAdminClient.listModels();
+System.out.println("We have following models in the account:");
+customDocumentModels.forEach(documentModelInfo -> {
+    System.out.printf("Model ID: %s%n", documentModelInfo.getModelId());
+    modelId.set(documentModelInfo.getModelId());
+
+    // get custom document analysis model info
+    DocumentModel documentModel = documentModelAdminClient.getModel(documentModelInfo.getModelId());
+    System.out.printf("Model ID: %s%n", documentModel.getModelId());
+    System.out.printf("Model Description: %s%n", documentModel.getDescription());
+    System.out.printf("Model created on: %s%n", documentModel.getCreatedOn());
+    documentModel.getDocTypes().forEach((key, docTypeInfo) -> {
+        docTypeInfo.getFieldSchema().forEach((field, documentFieldSchema) -> {
+            System.out.printf("Field: %s", field);
+            System.out.printf("Field type: %s", documentFieldSchema.getType());
+            System.out.printf("Field confidence: %.2f", docTypeInfo.getFieldConfidence().get(field));
+        });
+    });
+});
+
+// Delete Model
+documentModelAdminClient.deleteModel(modelId.get());
+```
+
+#### Analyze Documents using a Custom model
+Analyze data from custom documents using your own custom models, built with your own data, so they're tailored to your documents.
+In 3.x, `beginRecognizeCustomForms` or `beginRecognizeCustomFormsFromUrl` method was used to analyze receipts.
+In 4.X, `beginRecognizeCustomForms` and `beginRecognizeCustomFormsFromUrl` has been replaced with `beginAnalyzeDocument` and `beginAnalyzeDocumentFromUrl` respectively. Below is an example to analyze custom document data using 4.X `beginAnalyzeDocumentFromUrl`:
+<!-- embedme ./src/samples/java/com/azure/ai/formrecognizer/ReadmeSamples.java#L83-L102 -->
```

Comment:
I think it's worth describing the different return types - `AnalyzeResult` vs List of `RecognizedForm` and maybe different options here like `pages` now a string, no more `include_field_elements`, etc